### PR TITLE
Parameterized CUDALite MatMul

### PR DESCRIPTION
**Pull request by yodada**
In this PR:
 - Parameterized CUDALite MatMul w/ DMA
 - removed `-g` and `-ffp-contract=off` from c/cpp flags
 - added `-fno-rtti` to c/cpp flags

Generated assembly looks good, but we might want to double check in the near future. I think the compilation options are still not exactly the same as the ones used when having kernel code in bsg_manycore.

### CODE REVIEW

File: examples/cuda/riscv.mk [hunk 1]
```diff
@@ -212,11 +212,10 @@ RISCV_ARCH_OP     := rv32imaf
 # CCPPFLAGS are common between GCC and G++
 RISCV_CCPPFLAGS += $(RISCV_OPT_LEVEL)
 RISCV_CCPPFLAGS += -march=$(RISCV_ARCH_OP)
-RISCV_CCPPFLAGS += -g
+RISCV_CCPPFLAGS += -fno-rtti
```
Comment (mrutt92) [inline examples/cuda/riscv.mk]: see comment in other pr

File: examples/cuda/param-base-matmul/main.c [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+                 * Define tg_dim_x/y: number of tiles in each tile group
+                 * Calculate grid_dim_x/y: number of tile groups needed based on block_size_x/y
+                 ******************************************************************************************************************/
+                hb_mc_dimension_t tg_dim = { .x = bsg_tiles_X, .y = bsg_tiles_Y};
+                hb_mc_dimension_t grid_dim = { .x = 1, .y = 1};
+
+                /*****************************************************************************************************************
+                 * Prepare list of input arguments for kernel.
+                 ******************************************************************************************************************/
+                int cuda_argv[6] = {A_device, B_device, C_device, N, GRAIN_SIZE, dram_buffer};
```
Comment (mrutt92) [inline examples/cuda/param-base-matmul/main.c:213]: fun fact, you don't actually need the '6' here.

File: examples/cuda/param-base-matmul/main.c [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+                zero( C2, N );
+
+                print( A, N );
+                printf("----\n");
+                print( B, N );
+
+                /*****************************************************************************************************************
+                 * Copy A & B from host onto device DRAM.
+                 ******************************************************************************************************************/
+                hb_mc_dma_htod_t htod[2] = {{
```
Comment (mrutt92) [inline examples/cuda/param-base-matmul/main.c:191]: fun fact, you don't need the '2'

**Review by mrutt92 — APPROVED** [review]

**Comment by yodada** [comment]
wait something is not right ... I think I'm not merging that many files?

**Comment by mrutt92** [comment]
maybe you need to rebase?

**Comment by yodada** [comment]
I gonna do a manual rebase and create a new PR. This is not right. Gonna run for the flight

**Comment by mrutt92** [comment]
safe travels!

**Pull request closed without merging**